cocotb/__init__.py

```diff
@@ -126,6 +128,19 @@ def _initialise_testbench(root_name):
     """
     _rlock.acquire()
 
+    global argc, argv, SIM_NAME, SIM_VERSION
+    argc = argc_
+    argv = argv_
```

Comment:
What are your thoughts about setting `sys.argv` with these as part of `gpi_embed`?

Reply:
That might be useful if we were trying to make the entry module seem more like a script where we don't specify an entry function, but I'm not sure if that is best. Right now I'm seeing the entry function as more of a `main` function. We could (and probably should) move the simulator info to getters in the simulator module.

Reply:
> We could (and probably should) move the simulator info to getters in the simulator module and we can get rid of argc form the call.

Makes sense to me.

Reply:
Opened https://github.com/cocotb/cocotb/issues/1428 to discuss just putting this in `sys.argv`